### PR TITLE
Stop monkey patching the user agent, use the official customUserAgent

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Added tests
-   [ ] Code cleanup

## Checklist

-   [ ] I have read the **README** document
-   [ ] I have read the **CONTRIBUTING** document
-   [ ] Local run of `npm run fullBuild` succeeds
-   [ ] I have run tslint against my changes using the root tslint.yaml
-   [ ] My code follows the code style of this project
-   [ ] I have added tests to cover my changes
-   [ ] All new and existing tests passed
-   [ ] A short description of the change has been added to the **CHANGELOG** (CHANGELOG.md) if the change is feature related

## License

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Tasks/Common/sdkutils.ts [hunk 1]
```diff
@@ -26,32 +27,29 @@ export abstract class SdkUtils {
...
             console.warn(`Task manifest ${taskManifestFilePath} not found, cannot set custom user agent!`)
+
+            return undefined
         }
+        const agentVersion = process.env.AGENT_VERSION
+        const taskManifest = JSON.parse(fs.readFileSync(taskManifestFilePath, 'utf8')) as VSTSTaskManifest
+        const version = taskManifest.version
+        const userAgentString = `${this.userAgentPrefix}/${version.Major}.${version.Minor}.${version.Patch} ${
+            this.userAgentSuffix
+        }${agentVersion}-${taskManifest.name}`
```
Comment (awschristou) [inline Tasks/Common/sdkutils.ts]: PR #235 replaces the dash with a space.

File: Tasks/Common/sdkutils.ts [hunk 2]
```diff
@@ -143,26 +141,25 @@ export abstract class SdkUtils {
...
-        if (awsServiceOpts) {
-            if (!awsServiceOpts.credentials) {
-                const creds = await getCredentials(taskParams)
-                if (creds) {
-                    awsServiceOpts.credentials = await creds.getPromise()
-                }
-            }
-            if (!awsServiceOpts.region) {
-                awsServiceOpts.region = await getRegion()
+        if (!awsServiceOpts.credentials) {
```
Comment (awschristou) [inline Tasks/Common/sdkutils.ts]: Any chance awsServiceOpts is undefined?

File: Tasks/Common/sdkutils.ts (hunk 1 above)
Comment (hunterwerlla) [reply Tasks/Common/sdkutils.ts]: Yes it does, these do two different things and need to be merged

Comment (hunterwerlla) [reply Tasks/Common/sdkutils.ts]: It also completely changes the format  :P

File: Tasks/Common/sdkutils.ts (hunk 2 above)
Comment (hunterwerlla) [reply Tasks/Common/sdkutils.ts]: If it is, we should fail the task. It is declared non nullable and is only called from defaultClients.ts. All of the calls there always construct a service opts, and it never makes any sense to do this call without them.

**Comment by hunterwerlla** [comment]
I'm closing this and backlog the task

**Pull request closed without merging**